web_dashboard_tile/models/tile_tile.py

```diff
@@ -1,155 +1,127 @@
+
+    field_function = fields.Selection([
         ('min', 'Minimum'),
         ('max', 'Maximum'),
         ('sum', 'Sum'),
         ('avg', 'Average'),
         ('median', 'Median'),
         ], string='Function')
     field_id = fields.Many2one(
-        comodel_name='ir.model.fields', string='Field',
+        'ir.model.fields',
+        string='Field',
         domain="[('model_id', '=', model_id),"
         " ('ttype', 'in', ['float', 'int'])]")
-    active = fields.Boolean(
-        compute='_get_tile_info', readonly=True, search='_search_active')
-    background_color = fields.Char(default='#0E6C7E', oldname='color')
-    font_color = fields.Char(default='#FFFFFF')
-    sequence = fields.Integer(default=0, required=True)
+    helper = fields.Char(compute='_compute_function_helper')
+
+    @api.one
+    def _compute_data(self):
+        self.count = 0
+        self.computed_value = 0
+        # Compute count item
+        model = self.env[self.model_id.model]
+        eval_context = self._get_eval_context()
+        self.count = model.search_count(eval(self.domain, eval_context))
+        # Compute datas for field_id depending of field_function
+        if self.field_function and self.field_id and self.count != 0:
+            records = model.search(eval(self.domain, eval_context))
+            vals = [x[self.field_id.name] for x in records]
+            if self.field_function == 'min':
+                self.computed_value = min(vals)
+            elif self.field_function == 'max':
+                self.computed_value = max(vals)
+            elif self.field_function == 'sum':
+                self.computed_value = sum(vals)
+            elif self.field_function == 'avg':
+                self.computed_value = sum(vals) / len(vals)
+            elif self.field_function == 'median':
+                self.computed_value = self.median(vals)
+
+    @api.one
+    @api.onchange('field_function', 'field_id')
+    def _compute_function_helper(self):
+        self.helper = ''
+        if self.field_function and self.field_id:
+            desc = self.field_id.field_description
+            helpers = {
+                'min': "Minimum value of '%s'",
+                'max': "Maximum value of '%s'",
+                'sum': "Total value of '%s'",
+                'avg': "Average value of '%s'",
+                'median': "Median value of '%s'",
+            }
+            self.helper = _(helpers.get(self.field_function, '')) % desc
+
+    # Constraints and onchanges
+    @api.one
+    @api.constrains('model_id', 'field_id')
+    def _check_model_id_field_id(self):
+        if self.field_id and self.field_id.model_id.id != self.model_id.id:
+            raise ValidationError(
+                _("Please select a field of the selected model."))
+
+    @api.one
+    @api.constrains('field_id', 'field_function')
+    def _check_field_id_field_function(self):
+        if self.field_id and not self.field_function or\
```

Comment:
Admittedly somewhat confused by this if statement, but I think this hits it and makes slightly more clear:

```
validations = self.field_id, self.field_function
if any(validations) and not all(validations):
```